### PR TITLE
feat(docs): New Reference documentation section

### PR DESCRIPTION
**Pull request by fhennig**
# Description

feat(docs): New Reference documentation section

depends on this PR: https://github.com/stackabletech/documentation/pull/511

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/style-guide).
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: docs/modules/druid/partials/nav.adoc [hunk 1]
```diff
@@ -18,3 +16,9 @@
 *** xref:druid:usage-guide/operations/pod-placement.adoc[]
 *** xref:druid:usage-guide/operations/pod-disruptions.adoc[]
 *** xref:druid:usage-guide/operations/graceful-shutdown.adoc[]
+* xref:druid:reference/index.adoc[]
+** xref:druid:reference/crds.adoc[]
+*** {crd-docs}/druid.stackable.tech/druidcluster/v1alpha1/[DruidCluster]
```
Comment (sbernauer) [inline docs/modules/druid/partials/nav.adoc:21]: Hard-coding `v1alpha1` can introduce problems in the future once we start versioning CRDs. E.g. druid-op 24.x can handle DruidCluster v1alpha1, v1beta1, `v1 and v2. We would avoid the problem by only linking to https://crds.stackable.tech/druid-operator/nightly/

**Review by sbernauer — COMMENTED** [review]
Looks good in general!

File: docs/modules/druid/partials/nav.adoc (hunk 1 above)
Comment (fhennig) [reply docs/modules/druid/partials/nav.adoc:21]: I'd say we leave this for now? Everything is versioned. So we can just change the link once we change our CRD versions too. This link will always be valid in the future.

Comment (fhennig) [reply docs/modules/druid/partials/nav.adoc:21]: I would really like to link direclty to the DruidCluster. Once we version things, I'd always link to the stable/most recent version of the CRD.

Comment (fhennig) [reply docs/modules/druid/partials/nav.adoc:21]: Or maybe once we have a lot of version we stop linking to the object, but at least for now I think it's quite useful and I'd leave it in

Comment (sbernauer) [reply docs/modules/druid/partials/nav.adoc:21]: Works for me

**Review by sbernauer — APPROVED** [review]
Thanks!

**Pull request merged**